apps/ex_wire/lib/ex_wire/sync.ex

```diff
@@ -97,47 +130,165 @@ defmodule ExWire.Sync do
     {:noreply, handle_block_bodies(block_bodies, state)}
   end
 
+  def handle_info({:packet, %SnapshotManifest{} = snapshot_manifest, peer}, state) do
+    {:noreply, handle_snapshot_manifest(snapshot_manifest, peer, state)}
+  end
+
+  def handle_info({:packet, %SnapshotData{} = snapshot_data, peer}, state) do
+    {:noreply, handle_snapshot_data(snapshot_data, peer, state)}
+  end
+
   def handle_info({:packet, packet, peer}, state) do
     :ok = Exth.trace(fn -> "[Sync] Ignoring packet #{packet.__struct__} from #{peer}" end)
 
     {:noreply, state}
   end
 
   @doc """
-  Dispatches a packet of `GetBlockHeaders` to all peers for the next block
+  Dispatches a packet of `GetSnapshotManifest` to all capable peers.
+
+  # TODO: That "capable peers" part.
```

Comment:
Again this is mostly so I understand the meaning here; "capable peers" are peers that tell us they are able to receive the snapshot_manifest, correct? Will peers explicitly ask for snapshot_manifests or is the expectation that once we've setup a connection with peers that we'll keep sending them snapshot_manifests if they are capable?

Reply:
Peers advertise capabilities during the handshake process. In this case, we should send a request to all peers with `{'par', 1}` capabilities. Finding and sending to the right peers is kind of the easy part, dealing with multiple manifests and coming to a sane way of processing them will be the hard part (maybe we take the best and still farm out requests to multiple peers? what happens when we're started a sync and then get a better manifest? etc)